service-workers/service-worker/navigation-preload/get-state.https.html

```diff
@@ -14,204 +14,159 @@
 
-  return service_worker_unregister_and_register(t, script, scope)
-    .then(r => {
-        registration = r;
-        add_completion_callback(() => registration.unregister());
-        worker = registration.installing;
-        return wait_for_state(t, worker, 'activated');
-      })
-    .then(() => {
-        // Call getState().
-        return post_and_wait_for_reply(worker, 'getState');
-      })
-    .then(data => {
-        return Promise.all([data, registration.navigationPreload.getState()]);
-      })
-    .then(states => {
-        expect_navigation_preload_state(states[0], false, 'true',
-                                        'default state (from worker)');
-        expect_navigation_preload_state(states[1], false, 'true',
-                                        'default state (from page)');
-        // Call enable() and then getState().
-        return post_and_wait_for_reply(worker, 'enable');
-      })
-    .then(data => {
-        assert_equals(data, undefined, 'enable() should resolve to undefined');
-        return Promise.all([
-            post_and_wait_for_reply(worker, 'getState'),
-            registration.navigationPreload.getState()
-          ]);
-      })
-    .then(states => {
-        expect_navigation_preload_state(states[0], true, 'true',
-                                        'state after enable() (from worker)');
-        expect_navigation_preload_state(states[1], true, 'true',
-                                        'state after enable() (from page)');
-        // Call disable() and then getState().
-        return post_and_wait_for_reply(worker, 'disable');
-      })
-    .then(data => {
-        assert_equals(data, undefined,
-                      '.disable() should resolve to undefined');
-        return Promise.all([
-            post_and_wait_for_reply(worker, 'getState'),
-            registration.navigationPreload.getState()
-          ]);
-      })
-    .then(states => {
-        expect_navigation_preload_state(states[0], false, 'true',
-                                        'state after disable() (from worker)');
-        expect_navigation_preload_state(states[1], false, 'true',
-                                        'state after disable() (from page)');
-        return post_and_wait_for_reply(worker, 'setHeaderValue');
-      })
-    .then(data => {
-        assert_equals(data, undefined,
-                      '.setHeaderValue() should resolve to undefined');
-        return Promise.all([
+  const registration =
+      await service_worker_unregister_and_register(t, SCRIPT, SCOPE);
+
+  add_completion_callback(async() => {
+    if (registration)
+      await registration.unregister();
+  })
+
+  const worker = registration.installing;
+  await wait_for_state(t, worker, 'activated');
+  let data = await post_and_wait_for_reply(worker, 'getState');
+  let states = await Promise.all([data,
+                                 registration.navigationPreload.getState()]);
```

Comment:
I'd prefer `registration` to line up with `data` rather than the opening brace.

It's personal preference, but I tend to format like the Google style guide:
"Multiline array initializers and object initializers are indented 2 spaces, with the braces on their own line, just like blocks."

So this becomes:
```
let states = await Promise.all([
  data,
  registration.navigationPreload.getState()
]);
```

You might also just run all JavaScript through prettier.io and use that.

Reply:
Thanks for your recommendation, I have updated all of code in latest commit, PTAL.